### PR TITLE
feat: upgrade to postcss@8

### LINKED ISSUE #10: support PostCSS 8?
The `cssnano-preset-simple` already releasing [v2.0.0](https://github.com/Timer/cssnano-preset-simple/releases/tag/v2.0.0) with PostCSS 8 support https://github.com/Timer/cssnano-preset-simple/commit/42a4d113af03d526fda8cc7100c4fd768a7d9f85
Do we have plan to upgrade the `cssnano-simple` also?

Mentioned on https://github.com/vercel/next.js/pull/17415#pullrequestreview-511712472

### LINKED ISSUE #23620: Wrong PostCSS version installed
### What version of Next.js are you using?

10.1.2

### What version of Node.js are you using?

14.16.0

### What browser are you using?

build issue

### What operating system are you using?

Mac Big Sur

### How are you deploying your application?

locally

### Describe the Bug

During a fresh install of `next.js` 2 versions of `postcss` are installed. `next.js` depends on v8 but v7 ends up in my node_modules

### Expected Behavior

Only 1 version of postcss is installed or the correct version is installed at the root

### To Reproduce

`npx create-next-app`

### PR DESCRIPTION
**Pull request by stefanprobst**
closes #10
closes https://github.com/vercel/next.js/issues/23620

this updates postcss to v8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,13 +20,16 @@
...
     "pretty-quick": "^3.1.0",
     "semantic-release": "^17.3.7"
   },
   "dependencies": {
-    "cssnano-preset-simple": "1.2.2",
-    "postcss": "^7.0.32"
+    "cssnano-preset-simple": "^2.0.0"
+  },
+  "peerDependencies": {
+    "postcss": "^8.2.2"
```
Comment (stefanprobst) [inline package.json:32]: v8.2.2 is needed (i think): see https://github.com/postcss/postcss/pull/1485

**Comment by kachkaev** [comment]
This would prevent Next.js apps from complaining about CVE-2021-23368. Snyk report: https://snyk.io/vuln/SNYK-JS-POSTCSS-1090595

File: package.json (hunk 1 above)
Comment (mazipan) [reply package.json:32]: https://github.com/Timer/cssnano-preset-simple/commit/42a4d113af03d526fda8cc7100c4fd768a7d9f85 is using `^8.2.8` @stefanprobst

File: src/index.js [hunk 2]
```diff
@@ -1,34 +1,7 @@
...
-    }
-  } else {
-    return Promise.resolve(plugin()(css, result));
-  }
-
-  return Promise.resolve();
-}
-
-module.exports = postcss.plugin('cssnano-simple', (opts = {}) => {
+module.exports = (opts = {}) => {
```
Comment (mazipan) [inline src/index.js:4]: The changes were too much.
We need to check all the functionality in my opinion.

**Comment by Timer** [comment]
This looks great to me, thanks!

**Review by Timer — APPROVED** [review]

**Pull request merged**